### PR TITLE
Added clippy and fixed warnings.  Primarily changed `usize`s for `u64`s.

### PR DESCRIPTION
**Pull request by Fraser999**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/chunk_store/13)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @madadam

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/chunk_store.rs [hunk 1]
```diff
@@ -187,22 +188,22 @@ impl ChunkStore {
                 };
                 Ok(dir_entries.filter_map(dir_entry_to_routing_name).collect())
             })
-            .unwrap_or(vec![])
+            .unwrap_or_else(|_| Vec::new())
```
Comment (madadam) [inline src/chunk_store.rs:191]: Just nitpicking, but don't think this is a good idea. `Vec::new()` is zero allocation, so essentially free. `unwrap_or_else` involves calling the lambda, which might have overhead. Although it's probably going to be optimized away anyway, not sure.

Comment (madadam) [inline src/chunk_store.rs:191]: Plus its making the code a teeny bit less readable.

Comment (madadam) [inline src/chunk_store.rs:191]: Oh, this seems to one of Clippy's lints. Guess it's fine then.

**Comment by madadam** [comment]
Reviewed 2 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/chunk_store/13)_

**Pull request merged**